wagtail_transfer/migrations/0003_permissions.py

```diff
@@ -0,0 +1,36 @@
+from django.db import migrations
+
+
+def create_import_permission(apps, schema_editor):
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    content_type = ContentType.objects.get(app_label="wagtailcore", model="page")
+    Permission = apps.get_model("auth", "Permission")
+    Permission.objects.get_or_create(
+        content_type=content_type,
+        codename="wagtailtransfer_can_import",
+        name="Can import pages from other sites",
```

Comment:
These permissions don't just apply to pages though, snippet classes can also be imported. It would be good to note that here (attaching it to the `Page` `ContentType` also feels slightly off here for that reason, though the only model transfer provides is `IDMapping`, which I suppose would still be usable and is a bit more generic. The alternative, of having granular import permissions for each model type definitely feels like overkill)

Reply:
I did begin to add duplicate snippet & page permissions, but felt it was way too much code considering the likelihood of anyone using it.

Wagtail creates a placeholder content type to attach Wagtail admin permissions: https://github.com/wagtail/wagtail/blob/571b9e1918861bcbe64a9005a4dc65f3a1fe7a15/wagtail/admin/migrations/0001_create_admin_access_permissions.py#L11 We could add a migration that does something similar, if you felt strongly about the meaningfulness of the model that the permission is attached to. Otherwise, `IDMapping` sounds pretty reasonable.

Reply:
Yep, in that case I think changing the wording and using `IDMapping` sounds fine.

Reply:
Fixed in e1de684

Reply:
The permission check tests are failing for me - I think they just need the permission name in the decorators to be changed to match.

Additionally, I'm now having trouble running the tests, due to this migration. I believe this is due to relying on the `ContentType` created in the same app and running multiple migrations sequentially in the test suite - as `ContentTypes` are usually created triggered by the `post_migrate` signal, it can fail to find them under these circumstances. You may have a better approach, but I was able to get this to work using:

```
from django.contrib.contenttypes.management import create_contenttypes
from django.db import migrations


def create_import_permission(apps, schema_editor):
    app_config = apps.get_app_config('wagtail_transfer')
    app_config.models_module = app_config.models_module or True
    create_contenttypes(app_config)
    ContentType = apps.get_model("contenttypes", "ContentType")
```
This ensures that the transfer contenttypes are created before you require them to exist.
Let me know if you're unable to reproduce the bug or have a better workaround!

Reply:
Gah. I wish I'd never asked for code review here. I skipped local testing earlier, because I was on another project and couldn't remember how I'd run this test suite. I've now run the tests locally.

Thanks for your suggestion with the migration. The `or True` bit feels a bit hacky, but I suppose we just need to tell Django that there are models here, rather than that the models must be already registered.

I have fixed the other test issues now. It was the app namespace of the created permission as you suspected. It looks much better belonging to the `wagtail_transfer` app.

I have also now checked the functionality of the two wagtail_hooks.py changes in my project, though there's no unit tests for them.